packages/wrangler/src/assets.ts

```diff
@@ -443,6 +443,42 @@ export function validateAssetsArgsAndConfig(
 				"Please remove the asset binding from your configuration file, or provide a Worker script in your configuration file (`main`)."
 		);
 	}
+
+	// Smart placement turned on when using assets
+	if (
+		config?.placement?.mode === "smart" &&
+		config?.assets?.experimental_serve_directly
+	) {
+		logger.warn(
+			"Using assets with smart placement enabled may result in poor performance."
+		);
+	}
+
+	// User Worker ahead of assets, but no assets binding provided
+	if (
+		"legacy" in args
```

Comment:
I don't think we should be testing for `"legacy" in args` here. `legacy` is meant for things like `legacyAssets` or `site`, which are not relevant for this particular use case. So I think we ca go with just:
```
if(config?.assets?.experimental_serve_directly === false && config?.assets?.binding) {...}
```

Reply:
Did a bit of println debugging, and wrangler dev definitely puts the serve_directly value into legacy args :( Without it, we won't error/warn for dev, only deploy

Reply:
yeah, this is very likely some context I'm missing here, since we're doing a similar thing here: https://github.com/cloudflare/workers-sdk/blob/77bd9a1f8e3975ca8e6d9cb99d2ea3c035ec15a7/packages/wrangler/src/assets.ts#L436-L447. Let's leave as is